docs/home/howto/HOSTNAMES.md

```diff
@@ -0,0 +1,40 @@
+# Hostnames
+
+Epiphany manages host names only for Azure as this is
+[required](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#computer_name) by Terraform.
+
+## Limitations
+
+Linux host name cannot exceed 64 characters in length.
+
+## Generation rule
+
+By default, it generates automatically using the following rules:
+
+1. If `hostname_domain_extension` is set in `epiphany-cluster` configuration:
+`<cluster_prefix>-<cluster_name>-<component_name>-vm-<index>.<hostname_domain_extension>`
+
+2. If `hostname_domain_extension` is not set in `epiphany-cluster` configuration:
+`<cluster_prefix>-<cluster_name>-<component_name>-vm-<index>`.
+
+## Changing component name
+
+---
+**NOTE**
+
+Changing this configuration leads to replacement of component's virtual machines during re-apply.
+
+---
+
+Usually component names are predefined, but there is an ability to change them in host names.
+There might be such a need when `hostname_domain_extension` is set, and it results in exceeding the length limit.
+For example, to change `kubernetes-master` to `k8s-cp`, use following config:
+
+```yaml
+---
+kind: infrastructure/virtual-machine
+provider: azure
+name: kubernetes-control-plane-machine
```

Comment:
🤔 shouldn't it be `kubernetes-master-machine`? 
https://github.com/epiphany-platform/epiphany/blob/develop/schema/azure/defaults/infrastructure/virtual-machine.yml#L74